test/model/src/driver/kv_workload_generator.cpp

```diff
@@ -234,86 +234,6 @@ kv_workload_generator::sequence_traversal::complete_one(sequence_state *s)
-            std::ostringstream err;
-            err << "The stable timestamp must not be smaller than the oldest timestamp: " << t
-                << " < " << oldest << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-        stable = t;
-    }
-
-    if (std::holds_alternative<operation::set_oldest_timestamp>(op)) {
-        timestamp_t t = std::get<operation::set_oldest_timestamp>(op).oldest_timestamp;
-        if (t < oldest) {
-            std::ostringstream err;
-            err << "The oldest timestamp went backwards: " << oldest << " -> " << t << " (sequence "
-                << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-        if (t > stable && stable != k_timestamp_none) {
-            std::ostringstream err;
-            err << "The oldest timestamp must not be later than the stable timestamp: " << t
-                << " > " << stable << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-        oldest = t;
-    }
-
-    if (std::holds_alternative<operation::prepare_transaction>(op)) {
-        timestamp_t t = std::get<operation::prepare_transaction>(op).prepare_timestamp;
-        if (t < stable) {
-            std::ostringstream err;
-            err << "Prepare timestamp is before the stable timestamp: " << t << " < " << stable
-                << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-    }
-
-    if (std::holds_alternative<operation::set_commit_timestamp>(op)) {
-        timestamp_t t = std::get<operation::set_commit_timestamp>(op).commit_timestamp;
-        if (t < stable) {
-            std::ostringstream err;
-            err << "Commit timestamp is before the stable timestamp: " << t << " < " << stable
-                << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-    }
-
-    if (std::holds_alternative<operation::commit_transaction>(op)) {
-        timestamp_t t = std::get<operation::commit_transaction>(op).commit_timestamp;
-        if (t < stable) {
-            std::ostringstream err;
-            err << "Commit timestamp is before the stable timestamp: " << t << " < " << stable
-                << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-        t = std::get<operation::commit_transaction>(op).durable_timestamp;
-        if (t < stable && t != k_timestamp_none) {
-            std::ostringstream err;
-            err << "Durable timestamp is before the stable timestamp: " << t << " < " << stable
-                << " (sequence " << sequence.seq_no() << ")" << std::endl;
-            throw model_exception(err.str());
-        }
-    }
-}
-
 /*
  * kv_workload_generator::assign_timestamps --
```

Comment:
An efficiency improvement
Change new lines 287 - 294 from if; if; if; if; to if ; else if ; else if; else if;
This also makes the code read better since it now says explicitely that only one of these can be true.

Reply:
Yes, that sounds like a good idea. Fixed in [4a04b8e](https://github.com/wiredtiger/wiredtiger/pull/10553/commits/4a04b8e04991b6762a071383990aee502075a4d7).